packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -454,16 +388,43 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		storageAddOnsForPlan,
 	} );
 
-	const onCtaClick = useActionCallback( {
+	const priceString = formatCurrency(
+		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
+		currencyCode || 'USD',
+		{
+			stripZeros: true,
+			isSmallestUnit: true,
+		}
+	);
+
+	const { callback, text, postButtonText, status } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
 		planSlug,
+		planTitle,
+		priceString,
 		cartItemForPlan,
 		selectedStorageAddOn,
 	} );
 
-	// TODO: Unsure about using free plan as a fallback. We should revisit.
-	const onFreeTrialCtaClick = useActionCallback( {
+	const { callback: freeTrialCallback, text: freeTrialText } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		isFreeTrialAction: true,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
+		// TODO: Unsure about using free plan as a fallback. We should revisit.
 		planSlug: freeTrialPlanSlug ?? PLAN_FREE,
+		planTitle,
+		priceString,
 		cartItemForPlan: { product_slug: freeTrialPlanSlug ?? PLAN_FREE },
+		selectedStorageAddOn,
+	} );
+
+	const onCtaClick = useActionCallback( {
```

Comment:
**Note:**

`useActionCallback` will be removed in favor of `useAction` once this PR is done